node/lib/util/cherry_pick_util.js

```diff
@@ -177,6 +178,63 @@ exports.containsUrlChanges = co.wrap(function *(repo, commit, baseCommit) {
     return false;
 });
 
+const populateLibgit2MergeBugData = co.wrap(function*(repo, data) {
+    if (data.submoduleShasInBases === undefined) {
+        const head = data.head;
+        const targetCommit = data.targetCommit;
+        const mergeBases = yield GitUtil.mergeBases(repo, head, targetCommit);
+        const submoduleShasInBases = {};
+        for (const base of mergeBases) {
+            const commit = yield NodeGit.Commit.lookup(repo, base);
+            const names = yield SubmoduleUtil.getSubmoduleNamesForCommit(
+                repo, commit);
+            const shas = yield SubmoduleUtil.getSubmoduleShasForCommit(repo,
```

Comment:
This is closer in that we populate the data lazily (only when encountering an add/add situation), but there's still no need to read the SHAs for all (thousands) of submodules, just the ones in an add/add scenario.

My recommendation: cache the result of `mergeBases`, but do the she-reading on-demand.